File: src/lib/api/ecommerce-review-reply.ts
/**
 * 电商差评回复 API
 *
 * 封装电商差评回复相关的 Tauri 命令调用
 */

import { safeInvoke } from "@/lib/dev-bridge";

/**
 * 电商差评回复请求参数
 */
export interface EcommerceReviewReplyRequest {
  /** 电商平台 */
  platform: "taobao" | "jd" | "pinduoduo";
  /** 差评链接 */
  reviewUrl: string;
  /** 回复语气 */
  tone: "polite" | "sincere" | "professional";
  /** 回复长度 */
  length: "short" | "medium" | "long";
  /** 自定义模板 (可选) */
  template?: string;
  /** AI 模型 (可选) */
  model?: string;
  /** 执行 ID (可选) */
  executionId?: string;
}

/**
 * Skill 执行结果
 */
export interface SkillExecutionResult {
  /** 是否成功 */
  success: boolean;
  /** 最终输出 */
  output?: string;
  /** 错误信息 */
  error?: string;
  /** 已完成的步骤结果 */
  stepsCompleted: Array<{
    stepId: string;
    stepName: string;
    success: boolean;
    output?: string;
    error?: string;
  }>;
}

/**
 * 电商差评回复 API
 */
export const ecommerceReviewReplyApi = {
  /**
   * 执行电商差评回复
   *
   * @param request - 请求参数
   * @returns 执行结果
   */
  async executeReviewReply(
    request: EcommerceReviewReplyRequest
  ): Promise<SkillExecutionResult> {
    return safeInvoke(
      "execute_ecommerce_review_reply",
      request as unknown as Record<string, unknown>
    );
  },
};

export default ecommerceReviewReplyApi;

File: src-tauri/crates/skills/src/ecommerce_review_reply.rs
//! 电商差评回复 Skill
//!
//! 实现电商差评自动回复功能,支持淘宝、京东、拼多多等平台

use serde::{Deserialize, Serialize};

/// 电商平台类型
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum EcommercePlatform {
    /// 淘宝/天猫
    Taobao,
    /// 京东
    JD,
    /// 拼多多
    Pinduoduo,
}

impl EcommercePlatform {
    /// 获取平台名称
    pub fn name(&self) -> &str {
        match self {
            EcommercePlatform::Taobao => "淘宝",
            EcommercePlatform::JD => "京东",
            EcommercePlatform::Pinduoduo => "拼多多",
        }
    }

    /// 获取平台的回复风格特点
    pub fn reply_style(&self) -> &str {
        match self {
            EcommercePlatform::Taobao => "亲切、友好、注重客户体验",
            EcommercePlatform::JD => "专业、高效、强调服务保障",
            EcommercePlatform::Pinduoduo => "热情、实惠、突出性价比",
        }
    }
}

/// 回复配置
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReplyConfig {
    /// 回复语气: polite(礼貌), sincere(真诚), professional(专业)
    pub tone: String,
    /// 回复长度: short(简短), medium(中等), long(详细)
    pub length: String,
    /// 回复模板类型
    pub template: Option<String>,
}

impl Default for ReplyConfig {
    fn default() -> Self {
        Self {
            tone: "sincere".to_string(),
            length: "medium".to_string(),
            template: None,
        }
    }
}

/// 电商差评回复请求
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EcommerceReviewReplyRequest {
    /// 电商平台
    pub platform: EcommercePlatform,
    /// 差评链接
    pub review_url: String,
    /// 回复配置
    pub config: ReplyConfig,
}

/// 电商差评回复结果
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EcommerceReviewReplyResult {
    /// 是否成功
    pub success: bool,
    /// 提取的差评内容
    pub review_content: Option<String>,
    /// 识别的问题类型
    pub problem_type: Option<String>,
    /// 生成的回复
    pub reply: Option<String>,
    /// 错误信息
    pub error: Option<String>,
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_platform_name() {
        assert_eq!(EcommercePlatform::Taobao.name(), "淘宝");
        assert_eq!(EcommercePlatform::JD.name(), "京东");
        assert_eq!(EcommercePlatform::Pinduoduo.name(), "拼多多");
    }

    #[test]
    fn test_reply_style() {
        assert!(EcommercePlatform::Taobao.reply_style().contains("亲切"));
        assert!(EcommercePlatform::JD.reply_style().contains("专业"));
        assert!(EcommercePlatform::Pinduoduo.reply_style().contains("热情"));
    }

    #[test]
    fn test_reply_config_default() {
        let config = ReplyConfig::default();
        assert_eq!(config.tone, "sincere");
        assert_eq!(config.length, "medium");
        assert!(config.template.is_none());
    }

    #[test]
    fn test_platform_serialization() {
        let platform = EcommercePlatform::Taobao;
        let json = serde_json::to_string(&platform).unwrap();
        assert_eq!(json, "\"taobao\"");

        let deserialized: EcommercePlatform = serde_json::from_str(&json).unwrap();
        assert_eq!(deserialized, EcommercePlatform::Taobao);
    }
}
